src/core/consent/createConsentStateMachine.js

```diff
@@ -13,6 +13,13 @@ governing permissions and limitations under the License.
 import { defer } from "../../utils";
 
 export const DECLINED_CONSENT = "The user declined consent.";
+export const DECLINED_CONSENT_ERROR_CODE = "declinedConsent";
+
+const createDeclinedConsentError = () => {
+  const error = new Error(DECLINED_CONSENT);
+  error.code = DECLINED_CONSENT_ERROR_CODE;
```

Comment:
Note that we add a code to the error. This is how, up in Core, we can segregate errors due to consent being declined versus other kinds of errors. There are a few ways we could have done this (e.g., using a different error "class" or detecting using a regex ran against the error's message instead of a checking a code), but I felt like this was a decent way of handling it.

Reply:
It is decent, and easy to extend and add other important codes in the future, and simple than sub classing. 👍